### PR TITLE
chore: initial repository config

### PR DESCRIPTION
**Pull request by paulobmarcos**
Initial repo configuration. 

Transpile TypeScript with Babel at the core for this PR.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2018 Protocol Labs Inc.
```
Comment (satazor) [inline LICENSE]: Change this to 2019

File: package.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  "bugs": {
+    "url": "https://github.com/ipfs-shipyard/js-idm-wallet/issues"
+  },
+  "files": [
+    "lib"
+  ],
+  "scripts": {
+    "typechecking": "tsc",
+    "test": "jest",
+    "build": "tsc && babel src --out-dir lib --extensions \".ts\"",
```
Comment (satazor) [inline package.json]: Worth compiling to es too

**Review by satazor — CHANGES_REQUESTED** [review]

**Comment by satazor** [comment]
If TypeScript works well for us, we might consider adding it to MOXY’s eslint and Babel presets later on.

File: package.json (hunk 2 above)
Comment (paulobmarcos) [reply package.json]: After some thought we decided to not use TypeScript after all.

**Comment by paulobmarcos** [comment]
After some thought we decided to not use TypeScript after all. The transpile of TypeScript was not dealing private methods as we wanted and actually making them publicly available.

We are going to use [proposal-private-methods](https://github.com/tc39/proposal-private-methods), that reached [Stage 3](https://tc39.github.io/process-document/), together with [@babel/plugin-proposal-private-methods](https://babeljs.io/docs/en/next/babel-plugin-proposal-private-methods.html).

**Pull request merged**